pkg/agent/route/route_linux.go

```diff
@@ -326,84 +246,73 @@ func (c *Client) Reconcile(podCIDRs []string) error {
 		if desiredPodCIDRs.Has(entry) {
 			continue
 		}
-		klog.V(4).Infof("Deleting orphaned ip %s from ipset", entry)
+		klog.Infof("Deleting orphaned PodIP %s from ipset and route table", entry)
 		if err := ipset.DelEntry(antreaPodIPSet, entry); err != nil {
 			return err
 		}
+		_, cidr, err := net.ParseCIDR(entry)
+		if err != nil {
+			return err
+		}
+		route := &netlink.Route{Dst: cidr}
+		if err := netlink.RouteDel(route); err != nil && err != unix.ESRCH {
```

Comment:
Out of curiosity, would the routes be deleted by the following code if we don't do it here?

Reply:
RouteDelete here are for routes installed as part of podCIDR

RouteDelete follows are routes that are added to gw0 inadvertently by user.